### ISSUE TITLE
Plugins does not seem to work

### ISSUE THREAD

**Issue by qdm12** [abiosoft/caddy-docker#92]
Hi there,

First, thanks for the great work on that very useful container.

I am having troubles using plugins with the your container. 

1. I am building with:

    ```bash
     docker build -t mycaddy --build-arg plugins=filemanager github.com/abiosoft/caddy-docker.git
    ```

2. Running with:

    ```bash
    docker run -it --rm -p 443:443 -p 80:80 -p 37040:2015 -v /Caddyfile:/etc/Caddyfile:ro \
    -v /certificates:/root/.caddy mycaddy
    ```

3. I have a line `filemanager / /` in my Caddyfile (inside my domain name).

4. The container fails with the error: *no action found for directive 'filemanager' with server type 'http' (missing a plugin?)*

What am I missing here ?

Thanks !!

**Comment by abiosoft** [comment]
First verify if the plugin is present with `docker run -it --rm mycaddy -plugins | grep filemanager`

**Comment by qdm12** [comment]
It shows nothing.  I also tried with `docker build -t mycaddy --build-arg plugins=http.filemanager github.com/abiosoft/caddy-docker.git`.

Entering `docker run -it --rm mycaddy -plugins` gives the following:

```
Server types:
  http

Caddyfile loaders:
  short
  flag
  default

Other plugins:
  http.basicauth
  http.bind
  http.browse
  http.errors
  http.expvar
  http.ext
  http.fastcgi
  http.gzip
  http.header
  http.index
  http.internal
  http.limits
  http.log
  http.markdown
  http.mime
  http.pprof
  http.proxy
  http.push
  http.redir
  http.request_id
  http.rewrite
  http.root
  http.status
  http.templates
  http.timeouts
  http.websocket
  on
  shutdown
  startup
  tls
  tls.storage.file
```

I also tried with git and it does not appear in the list either unfortunately.

**Comment by abiosoft** [comment]
It ran successful for me and I can find filemanager in the list of the plugins. Are you sure it's not using a cache ? Try `--no-cache`.

**Comment by qdm12** [comment]
I retried by:
1. Removing all caddy related images
2. Removing dangling images with `docker image prune`
3. Building with `docker build -t mycaddy --no-cache --build-arg plugins=filemanager github.com/abiosoft/caddy-docker.git`

During the build, at step 11/18, the *filemanager* plugin is not present:
```
Step 11/18 : RUN /usr/bin/caddy -plugins
 ---> Running in 7538fe125434
Server types:
  http

Caddyfile loaders:
  short
  flag
  default

Other plugins:
  http.basicauth
  http.bind
  http.browse
  http.errors
  http.expvar
  http.ext
  http.fastcgi
  http.gzip
  http.header
  http.index
  http.internal
  http.limits
  http.log
  http.markdown
  http.mime
  http.pprof
  http.proxy
  http.push
  http.redir
  http.request_id
  http.rewrite
  http.root
  http.status
  http.templates
  http.timeouts
  http.websocket
  on
  shutdown
  startup
  tls
  tls.storage.file

 ---> dab5d13dec8f
Removing intermediate container 7538fe125434
```

4. Running `docker run -it --rm mycaddy -plugins | grep filemanager` which results in nothing still.

**Comment by abiosoft** [comment]
I'd be interested to see your output. Run with the following and share the contents of `out.log`.
```
docker build -t mycaddy --build-arg plugins=filemanager github.com/abiosoft/caddy-docker.git | tee out.log
```

**Comment by qdm12** [comment]
```
Sending build context to Docker daemon  155.6kB
Step 1/18 : FROM abiosoft/caddy:builder as builder
builder: Pulling from abiosoft/caddy
ff3a5c916c92: Already exists
f32d2ea73378: Pulling fs layer
3bdfb30a4c89: Pulling fs layer
6487ee6212c5: Pulling fs layer
074903419fc0: Pulling fs layer
f1b3b9beb708: Pulling fs layer
1c65c716f3ac: Pulling fs layer
f1b3b9beb708: Waiting
074903419fc0: Waiting
3bdfb30a4c89: Verifying Checksum
3bdfb30a4c89: Download complete
f32d2ea73378: Download complete
074903419fc0: Verifying Checksum
074903419fc0: Download complete
1c65c716f3ac: Verifying Checksum
f32d2ea73378: Pull complete
3bdfb30a4c89: Pull complete
f1b3b9beb708: Verifying Checksum
f1b3b9beb708: Download complete
6487ee6212c5: Verifying Checksum
6487ee6212c5: Download complete
6487ee6212c5: Pull complete
074903419fc0: Pull complete
f1b3b9beb708: Pull complete
1c65c716f3ac: Pull complete
Digest: sha256:6c70ba4a7f1d1bae56b2893380d859f6202538eaf5395168886b1ecc04ebb31c
Status: Downloaded newer image for abiosoft/caddy:builder
 ---> b7d9d7eefd34
Step 2/18 : ARG version="0.10.11"
 ---> Running in e2dd40634826
 ---> 3c3dedb1d3ca
Removing intermediate container e2dd40634826
Step 3/18 : ARG plugins="git"
 ---> Running in 8fc9db247934
 ---> 17967ae276f8
Removing intermediate container 8fc9db247934
Step 4/18 : RUN VERSION=${version} PLUGINS=${plugins} /bin/sh /usr/bin/builder.sh
 ---> Running in 2195eeab98a5
Cloning into '/go/src/github.com/mholt/caddy'...
Note: checking out 'd3f338ddab9ee24b376b5c9c51e202581e2d43ba'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

Switched to a new branch 'v0.10.11'
github.com/abiosoft/caddyplug (download)
github.com/abiosoft/errs (download)
github.com/fatih/color (download)
Fetching https://golang.org/x/sys/unix?go-get=1
Parsing meta tags from https://golang.org/x/sys/unix?go-get=1 (status code 200)
get "golang.org/x/sys/unix": found meta tag get.metaImport{Prefix:"golang.org/x/sys", VCS:"git", RepoRoot:"https://go.googlesource.com/sys"} at https://golang.org/x/sys/unix?go-get=1
get "golang.org/x/sys/unix": verifying non-authoritative meta tag
Fetching https://golang.org/x/sys?go-get=1
Parsing meta tags from https://golang.org/x/sys?go-get=1 (status code 200)
golang.org/x/sys (download)
github.com/fatih/color/vendor/github.com/mattn/go-isatty
golang.org/x/sys/unix
github.com/fatih/color/vendor/github.com/mattn/go-colorable
github.com/abiosoft/caddyplug/internal
github.com/abiosoft/errs
github.com/fatih/color
github.com/abiosoft/caddyplug/caddyplug
Cloning into '/go/src/github.com/caddyserver/builds'...
 ---> b9385b1f4d74
Removing intermediate container 2195eeab98a5
Step 5/18 : FROM alpine:3.7
 ---> 3fd9065eaf02
Step 6/18 : LABEL maintainer "Abiola Ibrahim <abiola89@gmail.com>"
 ---> Using cache
 ---> 0d3d4bd5e4bb
Step 7/18 : LABEL caddy_version "0.10.11"
 ---> Using cache
 ---> 161a2ab14358
Step 8/18 : RUN apk add --no-cache openssh-client git
 ---> Using cache
 ---> d39dbfa275b7
Step 9/18 : COPY --from=builder /install/caddy /usr/bin/caddy
 ---> Using cache
 ---> 62043a823898
Step 10/18 : RUN /usr/bin/caddy -version
 ---> Using cache
 ---> cb1cf5ae8573
Step 11/18 : RUN /usr/bin/caddy -plugins
 ---> Using cache
 ---> 037249601338
Step 12/18 : EXPOSE 80 443 2015
 ---> Using cache
 ---> 3191e6d6e700
Step 13/18 : VOLUME /root/.caddy /srv
 ---> Using cache
 ---> 4b111d6e760d
Step 14/18 : WORKDIR /srv
 ---> Using cache
 ---> 74d61f9869b2
Step 15/18 : COPY Caddyfile /etc/Caddyfile
 ---> Using cache
 ---> 7ddfa08959ed
Step 16/18 : COPY index.html /srv/index.html
 ---> Using cache
 ---> bc6e8198e8f4
Step 17/18 : ENTRYPOINT /usr/bin/caddy
 ---> Using cache
 ---> 356e5b6bdb5e
Step 18/18 : CMD --conf /etc/Caddyfile --log stdout
 ---> Using cache
 ---> 97203fb92dda
Successfully built 97203fb92dda
Successfully tagged mycaddy:latest
```

**Comment by abiosoft** [comment]
Kindly share the entire content of the log file. You can attach it or use a gist if you don't want to clutter this space.

**Comment by abiosoft** [comment]
This is where the problem lies. The arg is not getting recognised.
```
Step 3/18 : ARG plugins="git"
```

What is your docker version ?

**Comment by abiosoft** [comment]
Can you compare your log output to this ?

[out.log](https://github.com/abiosoft/caddy-docker/files/1820912/out.log)

**Comment by qdm12** [comment]
Thanks for the help ! I will try to debug it. My Docker version is **17.05.0-ce**. I'll let you know and close the issue once I found out.

**Comment by abiosoft** [comment]
I have to admit this is strange. The only obvious difference I can see is the build context size. Yours is `Sending build context to Docker daemon  155.6kB` while mine is just `87.04kB`.

I don't think this should have effect but maybe be explicit with the args.
```
 docker build -t mycaddy \
    --build-arg="plugins=filemanager" \
    https://github.com/abiosoft/caddy-docker.git
```

**Comment by qdm12** [comment]
It is definitely a problem with passing build-args.
I tried with a test Dockerfile:
```
FROM alpine
RUN echo ${x} # shows nothing
ARG x=test
RUN echo ${x} # shows nothing
ENV y=${x}
RUN echo ${y} # shows zzzz
```

with the build command `docker build -t mycaddy --build-arg x=zzzz ./`. The result seems strange to me or maybe I did not fully understand the use of build args.

Anyway, it must be a problem with Docker and not your repository. I found a workaround in case anyone needs it as well. Thanks for helping !
